src/gui/MainWindow.cc

```diff
@@ -2125,6 +2130,37 @@ void MainWindow::sendToPrintService()
 #endif // ifdef ENABLE_3D_PRINTING
 }
 
+void MainWindow::sendToCmd()
+{
+#ifdef ENABLE_3D_PRINTING
+  QString cmd = QString::fromStdString(Settings::Settings::cmdPrintCmd.value());
+
+  if (cmd.trimmed().isEmpty()) {
+    LOG(message_group::Error, Location::NONE, "", "Print cmd connection not configured. Please check preferences.");
+    return;
+  }
+
+  static QTemporaryFile exportFile{QDir::temp().filePath("OpenSCAD.XXXXXX.stl")};
+  if (!exportFile.open()) {
+    LOG(message_group::Error, Location::NONE, "", "Could not open temporary file.");
+    return;
+  }
+  const QString exportFilename = exportFile.fileName();
+
+  //Render the stl to a temporary file:
+  ExportInfo exportInfo = createExportInfo(FileFormat::STL, exportFilename, activeEditor->filepath);
+  exportFileByName(this->root_geom, exportInfo);
+
+  cmd.replace("#1", exportFilename);
```

Comment:
A simple replace is probably going to fail if there's spaces in the directory and/or filename.

Reply:
This should take care:
```diff
+  QStringList args = QProcess::splitCommand(cmd);
+  int var_idx = args.indexOf("#1");
+  if (var_idx >= 0) {
+    args.replace(var_idx, exportFilename);
+  }